jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -271,14 +273,28 @@ private void extractTar() {
   }
 
   private void assignLocalImageResult(Future<LocalImage> localImage) {
-    results.baseImageLayers = executorService.submit(() -> localImage.get().layers);
+
+    Future<List<Future<PreparedLayer>>> localBaseImageLayers =
+        executorService.submit(() -> localImage.get().layers);
+
     results.baseImageAndRegistryClient =
         executorService.submit(
-            () ->
-                LocalBaseImageSteps.returnImageAndRegistryClientStep(
-                        realizeFutures(results.baseImageLayers.get()),
-                        localImage.get().configurationTemplate)
-                    .call());
+            () -> {
+              return LocalBaseImageSteps.returnImageAndRegistryClientStep(
+                      realizeFutures(localBaseImageLayers.get()),
+                      localImage.get().configurationTemplate)
+                  .call();
+            });
+
+    results.baseImageLayers =
+        executorService.submit(
+            () -> {
+              Map<Image, List<Future<PreparedLayer>>> baseImageLayers = new HashMap<>();
+              baseImageLayers.put(
+                  results.baseImageAndRegistryClient.get().images.get(0),
+                  localBaseImageLayers.get());
```

Comment:
`localImage.get().layers` should work. No need to defined `localBaseImageLayers` on line 277.

and this will probably work:
```java
.submit( () -> Collections.singletonMap(
            results.baseImageAndRegistryClient.get().images.get(0),
            localImage.get().layers) );
```